Library/Homebrew/service.rb

```diff
@@ -0,0 +1,152 @@
+# typed: true
+# frozen_string_literal: true
+
+module Homebrew
+  # The {Service} class implements the DSL methods used in a formula's
+  # `service` block and stores related instance variables. Most of these methods
+  # also return the related instance variable when no argument is provided.
+  class Service
+    extend T::Sig
+    extend Forwardable
+
+    RUN_TYPE_IMMEDIATE = "immediate"
+    RUN_TYPE_INTERVAL = "interval"
+    RUN_TYPE_CRON = "cron"
+
+    # sig { params(formula: Formula).void }
+    def initialize(formula, &block)
+      @formula = formula
+      @run_type = RUN_TYPE_IMMEDIATE
+      @environment_variables = {}
+      @service_block = block
+    end
+
+    sig { params(command: T.nilable(T.any(T::Array[String], String, Pathname))).returns(T.nilable(Array)) }
+    def run(command = nil)
+      case T.unsafe(command)
+      when nil
+        @run
```

Comment:
Ideally this (and the other cases below) would have tests added. This is not blocking, I'd suggest it be done in a follow-up PR.